core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -208,10 +207,14 @@ private static void setupLogging(Settings settings) {
             e.printStackTrace();
         }
     }
-
-    private static Environment initialSettings(boolean foreground) {
-        Terminal terminal = foreground ? Terminal.DEFAULT : null;
-        return InternalSettingsPreparer.prepareEnvironment(EMPTY_SETTINGS, terminal);
+    private static Environment initialSettings(boolean daemonize, String pathHome, String pidFile) {
+        Terminal terminal = !daemonize ? Terminal.DEFAULT : null;
+        Settings.Builder builder = Settings.builder();
+        builder.put(Environment.PATH_HOME_SETTING.getKey(), pathHome);
+        if (!Strings.isNullOrEmpty(pidFile)) {
```

Comment:
This double-negative is kind of confusing, should be `if (Strings.hasText(pidFile)) {` instead, which also correctly handles strings like `" "`